src/Common/AzurePSCmdlet.cs

```diff
@@ -378,7 +380,16 @@ private void WriteBreakingChangeOrPreviewMessage()
                 && configManager.GetConfigValue<bool>(ConfigKeysForCommon.DisplayBreakingChangeWarning))
             {
                 BreakingChangeAttributeHelper.ProcessCustomAttributesAtRuntime(this.GetType(), this.MyInvocation, WriteWarning);
-                PreviewAttributeHelper.ProcessCustomAttributesAtRuntime(this.GetType(), this.MyInvocation, WriteDebug);
+
+                // Write preview message once for each cmdlet in one session
+                // Preview message may be outputed more than once if a cmdlet is concurrently called
+                // Considering lock affects performance but warning message is no harm
+                if (!AnnouncedPreviewMessageCmdlets.Contains(this.MyInvocation.MyCommand.Name)
```

Comment:
Better to check MyInvocation and MyCommand are not null, like
https://github.com/Azure/azure-powershell-common/blob/dfe6dba5490467c7855a497b0dfec04c5111f49e/src/Common/AzurePSCmdlet.cs#L683